### PR TITLE
net: lib: nrf_cloud: Fix nrf_cloud_obj_reset

### PR DESCRIPTION
**Pull request by glarsennordic**
Fix a small bug with error handling in nrf_cloud_obj_reset

Also add a couple of useful debug messages to nrf_cloud_codec.c

### CODE REVIEW

File: subsys/net/lib/nrf_cloud/src/nrf_cloud_codec_internal.c [hunk 1]
```diff
@@ -2321,6 +2321,9 @@ int nrf_cloud_cell_pos_req_json_encode(struct lte_lc_cells_info const *const inf
 		return -EINVAL;
 	}
 
+	LOG_DBG("Encoding lte_lc_cells_info with ncells_count: %d and gci_cells_count: %d",
```
Comment (jayteemo) [inline subsys/net/lib/nrf_cloud/src/nrf_cloud_codec_internal.c:2324]: unsigned?

File: subsys/net/lib/nrf_cloud/src/nrf_cloud_codec_internal.c [hunk 2]
```diff
@@ -2385,6 +2388,8 @@ int nrf_cloud_wifi_req_json_encode(struct wifi_scan_info const *const wifi,
 		return -EINVAL;
 	}
 
+	LOG_DBG("Encoding wifi_scan_info with count: %d", wifi->cnt);
```
Comment (jayteemo) [inline subsys/net/lib/nrf_cloud/src/nrf_cloud_codec_internal.c:2391]: unsigned?

**Review by jayteemo — APPROVED** [review]

**Review by plskeggs — APPROVED** [review]

File: subsys/net/lib/nrf_cloud/src/nrf_cloud_codec_internal.c (hunk 1 above)
Comment (glarsennordic) [reply subsys/net/lib/nrf_cloud/src/nrf_cloud_codec_internal.c:2324]: Doh

File: subsys/net/lib/nrf_cloud/src/nrf_cloud_codec_internal.c [hunk 3]
```diff
@@ -2321,6 +2321,9 @@ int nrf_cloud_cell_pos_req_json_encode(struct lte_lc_cells_info const *const inf
 		return -EINVAL;
 	}
 
+	LOG_DBG("Encoding lte_lc_cells_info with ncells_count: %d and gci_cells_count: %u",
```
Comment (glarsennordic) [inline subsys/net/lib/nrf_cloud/src/nrf_cloud_codec_internal.c:2324]: Ach, I missed another one. I cannot win with these unsigned prints

**Pull request merged**